wdl/AnnotateVcf.wdl

```diff
@@ -97,36 +97,52 @@ task SubsetVcf {
     RuntimeAttr? runtime_attr_override
   }
 
+  parameter_meta {
+    vcf: {
+      localization_optional: true
+    }
+    vcf_idx: {
+      localization_optional: true
+    }
+  }
+
   output {
     File subsetted_vcf     = "${prefix}.${contig}.vcf.gz"
     File subsetted_vcf_idx = "${prefix}.${contig}.vcf.gz.tbi"
   }
 
-  command <<<
-
-    set -euo pipefail
-
-    # Remote tabix to chromosome of interest
-    GCS_OAUTH_TOKEN=`gcloud auth application-default print-access-token` \
-      tabix -h ~{vcf} "~{contig}:0-300000000" | bgzip -c > "~{prefix}.~{contig}.vcf.gz"
-
-    tabix -p vcf -f "~{prefix}.~{contig}.vcf.gz"
-  
-  >>>
-  
   #########################
   RuntimeAttr default_attr = object {
-    cpu_cores:          1, 
-    mem_gb:             3.75, 
+    cpu_cores:          1,
+    mem_gb:             3.75,
     disk_gb:            50,
     boot_disk_gb:       10,
     preemptible_tries:  3,
     max_retries:        0
   }
   RuntimeAttr runtime_attr = select_first([runtime_attr_override, default_attr])
+
+  Float mem_gb = select_first([runtime_attr.mem_gb, default_attr.mem_gb])
+  Int java_mem_mb = ceil(mem_gb * 1000 * 0.8)
+
+  command <<<
+
+    set -euo pipefail
+
+    java -Xmx~{java_mem_mb}M -jar ${GATK_JAR} SelectVariants \
+      -V "~{vcf}" \
+      -L "~{contig}" \
+      -O ~{prefix}.~{contig}.vcf
+
+    # GATK does not block compress
```

Comment:
It should be block compressing -- is there a bug? I thought it would be setting up the output stream at `VariantContextWriterBuilder.determineOutputTypeFromFile`

Reply:
Apparently it depends on the way the output stream is written. I've fixed the tool so that it writes block compressed files now.